pkg/kubecost/allocationfilter_test.go

```diff
@@ -58,6 +58,36 @@ func Test_AllocationFilterCondition_Matches(t *testing.T) {
 
 			expected: false,
 		},
+		{
+			name: "ClusterID empty StartsWith '' -> true",
+			a: &Allocation{
+				Properties: &AllocationProperties{
+					Cluster: "",
+				},
+			},
+			filter: AllocationFilterCondition{
+				Field: FilterClusterID,
+				Op:    FilterStartsWith,
+				Value: "",
```

Comment:
Are we saying empty matches all? Your comment says `filterNamespaces="*"` which is different.

Reply:
For an `AllocationFilterCondition`, `Op: StartsWith` and `Value: ""` should match everything because all strings have a prefix of empty string. This is what `filterXYZ="*"` gets translated to by https://github.com/kubecost/opencost/pull/1211 and is the expected behavior.

Note that the `Op` isn't `FilterEquals`, but `FilterStartsWith`. Does that make sense?

Reply:
Yeah, so we are talking about external vs internal representation. Do we support the case of doing a wildcard anywhere?

Reply:
We do support wildcards right now in the external representation -- `filterNamespaces="*"` is a valid query in today's Allocation API, as is `filterNamespaces="kube-*"`. 

> Note: The v2 filter language does not yet support wildcards. It will support an equivalent query with something like `filter=namespace~:"kube-"`.